src/Biogeochemistry.jl

```diff
@@ -0,0 +1,177 @@
+module Biogeochemistry
+
+using Oceananigans.Grids: Center, xnode, ynode, znode
+using Oceananigans.Advection: div_Uc, CenteredSecondOrder
+using Oceananigans.Architectures: device, architecture
+
+import Oceananigans.Fields: CenterField
+
+#####
+##### Generic fallbacks for biogeochemistry
+#####
+
+"""
+Update tendencies.
+
+Called at the end of calculate_tendencies!
+"""
+update_tendencies!(bgc, model) = nothing
+
+"""
+Update tracer tendencies.
+
+Called at the end of calculate_tendencies!
+"""
+update_biogeochemical_state!(bgc, model) = nothing
+
+@inline biogeochemical_drift_velocity(bgc, val_tracer_name) = nothing
+@inline biogeochemical_advection_scheme(bgc, val_tracer_name) = nothing
+@inline biogeochemical_auxiliary_fields(bgc) = NamedTuple()
+
+"""
+AbstractBiogeochemistry.
+
+Abstract type for biogeochemical models. To define a biogeochemcial relaionship
+the following functions must have methods defined where `BiogeochemicalModel`
+is a subtype of `AbstractBioeochemistry`:
+ - `(bgc::BiogeochemicalModel)(i, j, k, grid, ::Val{:TRACER_NAME}, clock, fields)` which 
+    returns the biogeochemical reaction for for each tracer
+ - `required_biogeochemical_tracers(::BiogeochemicalModel)` which returns a tuple of
+    required tracer names
+ - `required_biogeochemical_auxiliary_fields(::BiogeochemicalModel)` which returns 
+    a tuple of required auxiliary fields
+ - `biogeochemical_auxiliary_fields(bgc::BiogeochemicalModel)` which returns a `NamedTuple`
+    of the models auxiliary fields (e.g. `(PAR = bgc.light_attenuation.PAR_field, )`)
+ - `biogeochemical_drift_velocity(bgc::BiogeochemicalModel, ::Val{:TRACER_NAME})` which 
+    returns a velocity fields (i.e. a `NamedTuple` of fields with keys `u`, `v` & `w`)
+    for each tracer
+ - `biogeochemical_advection_scheme(bgc::BiogeochemicalModel, ::Val{:TRACER_NAME})` which
+    returns an advection scheme for each tracer.
+ - `update_biogeochemical_state!(bgc::BiogeochemicalModel, model)` (optional) to update the
+    model state
+"""
```

Comment:
The formatting needs to be fixed

Reply:
How should it be for an absract type like this?

Reply:
Same indentation as for a docstring

Reply:
Tried to fix this now